plugins/package-managers/bundler/src/funTest/assets/projects/synthetic/lockfile/Gemfile

```diff
@@ -4,6 +4,7 @@ source "https://rubygems.org"
 
 gem 'nokogiri'
 gem 'rack', '~>2.1'
+gem 'psych'
```

Comment:
Can you help me understanding whether it is necessary to and why you choose to update the lockfiles in this same commit, as opposed to a separate commit?

Reply:
I guess the question is not about the lockfile actually, but why this dependency was added to the test `Gemfile` (implying lockfile updates) in the first place.

If you want to showcase the effect of your change at the example of this new dependency, IMO the dependency should be added in a preceding commit (incl. any necessary lockfile changes), and then a new commit should introduce your actual change and the necessary test assets changes.

So in short, separate the addition of new test dependencies from the effect of code changes.

Reply:
The intention here is that the psych gem is one example where the logic currently "breaks" in the sense that due to the order in which the different versions of the gem have been uploaded to RubyGems.org, fetching the data without an explicit `platform` parameter returns the Java version of it instead of the standard Ruby version.

Check these 2 requests:

- `curl 'https://rubygems.org/api/v2/rubygems/psych/versions/5.1.2.yaml'`
- `curl 'https://rubygems.org/api/v2/rubygems/psych/versions/5.1.2.yaml?platform=ruby'`

In contrast, other gems like nokogiri, which happen to have a different upload order, return the standard Ruby version first:

- `curl 'https://rubygems.org/api/v2/rubygems/nokogiri/versions/1.16.5.yaml'`
- `curl 'https://rubygems.org/api/v2/rubygems/nokogiri/versions/1.16.5.yaml?platform=java'`

According to RubyGems' own source code, [this (arguably non-deterministic) behavior is intentional](https://github.com/rubygems/rubygems.org/blob/master/app/models/rubygem.rb#L146-L159).

---

That being said: I had assumed that when adding the gem to the test `Gemfile`, some test in addition to the one that did fail (due to an additional gem being installed) would fail because of the change. But it didn't, so I could probably revert these changes.

Reply:
> The intention here is that the psych gem is one example where the logic currently "breaks"

Yeah, then this new dependency plus needed test assest chnages should be added in a separate commit before the commit that makes the code changes.

However...

> But it didn't, so I could probably revert these changes.

...if that sentence means there would be no `psych`-related changes to the test assets before and after your change, then I'd indeed drop the introduction of that dependency for simplicity.